### PR TITLE
Upgrade playbook

### PR DESCRIPTION
**Pull request by sdodson**
This playbook will update an existing OSE 3.0.0-3.0.1 install to 3.0.2. It will also upgrade Origin 1.0.2-1.0.5 to 1.0.6.

TODO: 
- [x] Check for default router / registry before attempting to upgrade
- [x] Update router / registry image name to handle origin/aos/etc

### CODE REVIEW

File: playbooks/adhoc/upgrades/upgrade.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  tasks:
+    - name: oadm policy reconcile-cluster-roles --confirm
+      command: >
+        {{ openshift.common.admin_binary}} --config=/etc/openshift/master/admin.kubeconfig
+        policy reconcile-cluster-roles --confirm
+
+- name: Upgrade default router
+  hosts: oo_first_master
+  vars:
+    - router_image: "{{ openshift.master.registry_url }} | replace( '${component}', 'router' ) | replace ( '${version}', v{{ _new_version.stdout }} ) "
```
Comment (sdodson) [inline playbooks/adhoc/upgrades/upgrade.yml]: @wshearn or @twiest Mind checking my templating here? This is evaluating to 

`openshift3/ose-${component}:${version} | replace( ${component}, router ) | replace ( ${version}, v3.0.2.0 ) `

I'm hoping for 
`openshift3/ose-router:v3.0.2.0`

registry_url is `openshift3/ose-${component}:${version}`

Comment (abutcher) [inline playbooks/adhoc/upgrades/upgrade.yml]: @sdodson Something like this... wrapping the whole thing in `{{ }}`

```
- router_image: "{{ openshift.master.registry_url | replace( '${component}', 'router' ) | replace ( '${version}', _new_version.stdout ) }}"
```

Comment (sdodson) [inline playbooks/adhoc/upgrades/upgrade.yml]: @abutcher thanks, that worked

**Comment by sdodson** [comment]
@wshearn Mind reviewing this? QE has validated functionality but they asked if perhaps this should be relocated to playbooks/byo/upgrades/ so a code review would be appreciated.

I'll squash things after review.

**Comment by sdodson** [comment]
@abutcher feedback welcome

File: playbooks/adhoc/upgrades/upgrade.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  tasks:
+    - debug: var=registry_image
+    - name: Check for default registry
+      command: >
+          {{ openshift.common.client_binary }} --config=/etc/openshift/master/admin.kubeconfig
+          get -n default dc/docker-registry
+      register: _default_registry
+      failed_when: false
+      changed_when: false
+    - name: Update router image to current version
```
Comment (abutcher) [inline playbooks/adhoc/upgrades/upgrade.yml]: s/router/registry

**Comment by abutcher** [comment]
@sdodson Aside from what we just talked with the config_dir and that inline comment lgtm :+1:

**Comment by sdodson** [comment]
@abutcher Thanks for taking a look, addressed your comments.

**Comment by wshearn** [comment]
I understand it is a tech preview, but could this read in the config already on disk(yaml file so I would think it is fairly easy to handle, just mapping the values) and so the config doesn't change just new values are laid down?

**Comment by sdodson** [comment]
@wshearn I don't know how we'd know which values should be overwritten and which shouldn't. Build the new file and merge all the values from the existing file? I'd prefer taking that as a follow up.

**Comment by sdodson** [comment]
@brenton I believe this is ready to merge provided we're OK with the caveat that it will overwrite configuration files creating backup files for review. This is noted in the README and in the openshift-docs PR https://github.com/openshift/openshift-docs/pull/1000

QE tested with that caveat and gave the go ahead to merge.

**Pull request merged**